cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -70,7 +70,7 @@ func NewCmdApply(apf *applyPlanFlags) *cobra.Command {
 		applyPlanFlags:   apf,
 		imagePullTimeout: defaultImagePullTimeout,
 		etcdUpgrade:      true,
-		criSocket:        kubeadmapiv1alpha3.DefaultCRISocket,
+		// Don't set criSocket to a default value here, as this will override the setting in the stored config in RunApply below.
```

Comment:
I think that we should deprecate/remove this flag now that kubeadm config / node annotation contains the cri to be used for this node.
This will lock down a possible failing case for updates (users passing wrong cri)

Reply:
It actually works even now in that way, we simply need to deprecate it for apply.

Reply:
we might want to resolve previous discussion in this PR?